controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/BadgeApiHandler.java

```diff
@@ -35,6 +40,7 @@
     private final static Logger log = Logger.getLogger(BadgeApiHandler.class.getName());
 
     private final Controller controller;
+    private final Map<Key, Value> badgeCache = new ConcurrentHashMap<>();
```

Comment:
Since we're adding this deter abuse we should probably enforce a maximum size?

Reply:
So each badge is ~5kb. If someone wants to play nasty, that's ~200Kb per job they know about, say ~1Mb per application. We don't have enough applications for this to be an issue. Also, if we were to reach the maximum size, someone could rotate the entries they were asking for, thus invalidating the whole cache anyway. So I'm not sure ... Perhaps we should actively evict entries expired entries, though, in case we get lots of badge serving in the future.